### PR TITLE
Update default HSTS max-age value to 1 year

### PR DESCRIPTION
**Pull request by grantbdev**
### Summary

This PR updates the default HSTS max-age value to 31536000 seconds (1 year) to meet the minimum max-age requirement for https://hstspreload.org/.

This requirement for the browser preload lists seemed to change in [October 2017](https://github.com/chromium/hstspreload.org/commit/b48ae5ac7c2db1d8a0635ee60a4c68a4b2f9c88e).

The HSTS max-age default was once set to 1 year, but it [was lowered](https://github.com/rails/rails/pull/21520/files#r39923850) because at that time 1 year was arbitrary and much more than the 18 week requirement for the browser preload lists. Assuming that the Rails defaults are intended to match the minimum requirements for the browser preload list, it makes sense to update this value. At the very least, the documentation could be updated to indicate that the requirements for the browser preload lists (at least for Chromium/https://hstspreload.org/) have changed.

### Other Information

I also noticed that the preload directive is required to be present in the header for https://hstspreload.org/. However, Rails does not add the directive by default. I do not know if it is the intention of the Rails defaults to meet all of the preload list requirements, but if it is then perhaps the existing preload directive option should be set to true.

### CODE REVIEW

File: actionpack/lib/action_dispatch/middleware/ssl.rb [hunk 1]
```diff
@@ -47,9 +47,8 @@ module ActionDispatch
   class SSL
     # :stopdoc:
 
-    # Default to 180 days, the low end for https://www.ssllabs.com/ssltest/
-    # and greater than the 18-week requirement for browser preload lists.
-    HSTS_EXPIRES_IN = 15552000
+    # Default to 1 year, the minimum for browser preload lists.
+    HSTS_EXPIRES_IN = 31536000
```
Comment (6temes) [inline actionpack/lib/action_dispatch/middleware/ssl.rb]: What about `1.year.to_i`?

File: actionpack/test/dispatch/ssl_test.rb [hunk 2]
```diff
@@ -98,8 +98,8 @@ def assert_post_redirected(redirect: {}, from: "http://a/b?c=d",
 end
 
 class StrictTransportSecurityTest < SSLTest
-  EXPECTED = "max-age=15552000"
-  EXPECTED_WITH_SUBDOMAINS = "max-age=15552000; includeSubDomains"
+  EXPECTED = "max-age=31536000"
+  EXPECTED_WITH_SUBDOMAINS = "max-age=31536000; includeSubDomains"
```
Comment (6temes) [inline actionpack/test/dispatch/ssl_test.rb]: Same here:

`"max-age=#{1.year.to_i}"` as in https://github.com/rails/rails/blob/3b4f14dc43f09addb50c546ffe23b9e03eafed20/railties/lib/rails/generators/rails/app/templates/config/environments/development.rb.tt#L22

File: actionpack/lib/action_dispatch/middleware/ssl.rb (hunk 1 above)
Comment (grantbdev) [reply actionpack/lib/action_dispatch/middleware/ssl.rb]: @6temes https://github.com/rails/rails/commit/5d149048ffa56c8fa1c41d1ef3bcf9472052f369

File: actionpack/test/dispatch/ssl_test.rb (hunk 2 above)
Comment (grantbdev) [reply actionpack/test/dispatch/ssl_test.rb]: @6temes Not sure if the intention behind https://github.com/rails/rails/commit/5d149048ffa56c8fa1c41d1ef3bcf9472052f369 applies in the tests as well.

File: actionpack/lib/action_dispatch/middleware/ssl.rb (hunk 1 above)
Comment (6temes) [reply actionpack/lib/action_dispatch/middleware/ssl.rb]: Makes sense...

**Comment by Fudoshiki** [comment]
Hello, what's blocking this PR?

**Comment by grantbdev** [comment]
@Fudoshiki I am not sure if there is anything blocking this PR, I just haven't heard from anyone about it.

@guilleiguaran Need me to rebase?

**Comment by guilleiguaran** [comment]
Thank you, I've merged and backported this to 5-2-stable.

**Pull request merged**